instrumentation/httpurlconnection/testing/src/androidTest/java/io/opentelemetry/instrumentation/library/httpurlconnection/InstrumentationTest.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.library.httpurlconnection;
+
+import static org.junit.Assert.assertEquals;
+
+import io.opentelemetry.android.test.common.OpenTelemetryTestUtils;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.sdk.testing.exporter.InMemorySpanExporter;
+import io.opentelemetry.sdk.trace.data.SpanData;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+public class InstrumentationTest {
+    private static final InMemorySpanExporter inMemorySpanExporter = InMemorySpanExporter.create();
+
+    @Before
+    public void setUp() {
+        OpenTelemetryTestUtils.setUpSpanExporter(inMemorySpanExporter);
+    }
+
+    @After
+    public void tearDown() {
+        inMemorySpanExporter.reset();
```

Comment:
That's a great point! Will bring it up for discussion in our next Android Sig meeting. :)

Reply:
not a blocker though, I believe more tests do this (global state).

Reply:
@marandaneto @LikeTheSalad Do you see any concerns with having an inMemorySpanExporter per test case? We can do that right away right now to avoid issues later if we were to run tests in parallel? I can make the change in okhttp3.InstrumentationTest.java too for same in this PR.

Reply:
if tests are not running in parallel, all good, if they do, it is a must to avoid concurrency issues and flaky tests

Reply:
I would not recommend putting any test infrastructure in static scope. At minimum, have one per test instance, but ideally you create a new one per test run if it involves additional threads. This project isn't so big that reinits are goign to slow the test suite runs down, so it's probably better to be as clean as possible to avoid weird flakiness.

Reply:
I agree, I think we should avoid statics in general unless it's necessary for a particular test case.

Reply:
I’ve added a new commit which addresses the following:

- Makes a per test instance of inMemorySpanExporter - thereby making it possible to run these tests in parallel
- For test purposes only - providing a function for re-setting the instrumenter instance for each test - thereby allowing us to re-set opentelemetry instance (with new exporter) for each test and not depending on GlobalOpentelemetry instance (If tests are run in parallel but in same JVM process, GlobalOpentelemetry instance would be shared, so we can’t use that). 

Some other considerations: 
- For now, for general use case, still depending on GlobalOpentelemetry instance until we have a way around that. (We cannot use a installer like AndroidInstrumentation.kt as we do not want to introduce the requirement of client having to make a manual call to install(opentelemetryInstance) after initializing their OpentelemetryInstance. That adds manual parts to this auto-instrumentation. And if we were to automate this install call, it would be difficult/tedious for auto-instrumentations to figure out where exactly to place this call in client code as it needs to be after their opentelemetryInstance initialization which could vary for each customer.)
- For now, I have not used CachedSupplier for getting the instrumenter instance lazily as it synchronizes every read access to instrumenter which I believe has more performance downside. Instead I’ve only synchronized the read access until instrumenter is null and made the variable volatile. We can discuss this more and establish the best approach and unify them. 

I can also make the change for okhttp3 androidTests, if this approach seems okay.